Assets/Shopify/UIToolkit/Theme Controllers/SingleProductThemeController.cs

```diff
@@ -25,6 +25,7 @@ public class SingleProductThemeController : ThemeControllerBase {
         }
 
         public override void OnShow() {
+            Theme.OnLoadingStarted();
```

Comment:
wondering if we should hook these events into all network requests that the controller makes, including when its creating a checkout for example? WDYT?

Reply:
I feel like from a UI perspective you may need to know what type of load was started and finished. Possible implementations:

* Pass an enum to the handler function
* Have separate handler functions to implement: `OnLoadingProductStart`, `OnLoadingCheckoutStart`, etc.

As I'm working on this I do wonder how often developers would just not implement certain functions.

Reply:
I like this approach, its what was outlined in the tech design:

```
OnLoadingStarted(LoadingContext context) -> void

Callback that is called when the controller begins loading something.  This callback exists so that the developer can show spinners or other loading views when pieces of the theme are loading. The context contains information about what the controller started loading, specifically an enum containing the different kinds of loading that an controller can perform. The possible values are ProductList, ProductDetails, Collections and Cart.
```